torch/fx/experimental/symbolic_shapes.py

```diff
@@ -800,29 +800,71 @@ def render(self, source: Source):
 DimConstraint = Union[StrictMinMaxConstraint, RelaxedUnspecConstraint, None]
 
 @dataclass(frozen=True)
```

Comment:
This appears to be frozen but not actually (because the union find is mutable and stored here). It would be nice to explain exactly in what sense this is frozen.

Reply:
It is frozen after post_init, i.e., once you pass in the public fields and we're done with initialization (which still doesn't touch the public fields, but does populate a couple of internal fields `_parents` and `_defs`), none of the fields are expected to change by code (even though the fields themselves are lists and dicts which are technically mutable).